src/TestableIO.System.IO.Abstractions.Wrappers/DirectoryAclExtensions.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Runtime.Versioning;
+using System.Security.AccessControl;
+
+namespace System.IO.Abstractions
+{
+    /// <summary>
+    ///     ACL (access control list) extension methods for <see cref="IDirectory" />.
+    /// </summary>
+    public static class DirectoryAclExtensions
+    {
+#if FEATURE_FILE_SYSTEM_ACL_EXTENSIONS
+        /// <inheritdoc cref="System.IO.FileSystemAclExtensions.CreateDirectory(DirectorySecurity, string)" />
+#else
+        /// <inheritdoc cref="Directory.CreateDirectory(string, DirectorySecurity)"/>
+#endif
+        [SupportedOSPlatform("windows")]
+        public static void CreateDirectory(this IDirectory directory,
+            string path,
+            DirectorySecurity directorySecurity)
+        {
+            IDirectoryInfo directoryInfo =
+                directory.FileSystem.DirectoryInfo.New(path);
+            IFileSystemExtensibility extensibility =
+                directoryInfo.Extensibility;
+            if (extensibility.TryGetWrappedInstance(out DirectoryInfo di))
+            {
+                di.Create(directorySecurity);
+            }
+            else
+            {
+                extensibility.StoreMetadata("AccessControl:DirectorySecurity",
+                    directorySecurity);
+                directoryInfo.Create();
+            }
```

Comment:
I don't fully understand why we need this code path. Couldn't we just throw a `NotSupportedException` when the underlying type does not support the ACL APIs?

Reply:
I need to store the `DirectoryInfo` so that I can return the value when calling `GetAccessControl` on a mocked instance.
The idea is to
- Use the functionality of the underlying file system when working on a real file system
- Otherwise store the value as metadata, so that I can retrieve it in other methods.

Reply:
> Otherwise store the value as metadata, so that I can retrieve it in other methods.

But in the end ACLs only work on the real file system, so why not prevent changing them in the beginning?